### PR TITLE
[Feature] [connector] Optimize doris connector checkConfig

### PR DESCRIPTION
**Pull request by simon824**
## Purpose of this pull request
subTask of #1153
Optimize doris connector checkConfig

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connector-spark-doris/src/main/scala/org/apache/seatunnel/spark/sink/Doris.scala [hunk 1]
```diff
@@ -68,35 +69,23 @@ class Doris extends SparkBatchSink with Serializable {
...
   override def checkConfig(): CheckResult = {
-    val checkResult =
-      checkAllExists(config, Config.HOST, Config.DATABASE, Config.TABLE_NAME, Config.USER, Config.PASSWORD)
+    checkAllExists(config, HOST, DATABASE, TABLE_NAME, USER, PASSWORD)
+  }
 
-    if (!checkResult.isSuccess) {
-      checkResult
-    } else if (config.hasPath(Config.USER) && !config.hasPath(Config.PASSWORD) || config.hasPath(
-        Config.PASSWORD) && !config.hasPath(Config.USER)) {
```
Comment (simon824) [inline seatunnel-connectors/seatunnel-connector-spark-doris/src/main/scala/org/apache/seatunnel/spark/sink/Doris.scala:77]: This judgement is redundant.

File: seatunnel-connectors/seatunnel-connector-spark-doris/src/main/scala/org/apache/seatunnel/spark/sink/Doris.scala [hunk 2]
```diff
@@ -68,35 +69,23 @@ class Doris extends SparkBatchSink with Serializable {
...
-      this.apiUrl = s"http://$host/api/$dataBase/$tableName/_stream_load"
-      if (TypesafeConfigUtils.hasSubConfig(config, Config.ARGS_PREFIX)) {
-        val properties = TypesafeConfigUtils.extractSubConfig(config, Config.ARGS_PREFIX, true)
-        val iterator = properties.entrySet().iterator()
-        while (iterator.hasNext) {
-          val map = iterator.next()
-          val split = map.getKey.split("\\.")
-          if (split.size == 2) {
-            propertiesMap.put(split(1), String.valueOf(map.getValue.unwrapped))
-          }
```
Comment (simon824) [inline seatunnel-connectors/seatunnel-connector-spark-doris/src/main/scala/org/apache/seatunnel/spark/sink/Doris.scala:92]: move these from `checkConfig()` to `prepare()`

**Comment by CalvinKirs** [comment]
@xtr1993 PTAL,Thx

**Comment by xtr1993** [comment]
> @xtr1993 PTAL,Thx

LGMT

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**